src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -361,6 +369,8 @@ def remove_extension(extension_name):
             raise CLIError(
                 "Extension '{name}' was installed in development mode. Remove using "
                 "`azdev extension remove {name}`".format(name=extension_name))
+        if is_cloud_shell_system_extension(ext.path):
+            raise CLIError(f"Cannot remove system extension {extension_name} in Cloud Shell.")
```

Comment:
So is there a way for customer to remove the extension?

Reply:
No, they don't have permission to remove system extension in Cloud Shell.